iis-common/src/main/java/eu/dnetlib/iis/common/fault/FaultUtils.java

```diff
@@ -0,0 +1,61 @@
+package eu.dnetlib.iis.common.fault;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import eu.dnetlib.iis.audit.schemas.Cause;
+import eu.dnetlib.iis.audit.schemas.Fault;
+
+/**
+ * {@link Fault} related utilities.
+ * @author mhorst
+ *
+ */
+public class FaultUtils {
+
+	/**
+	 * Generates {@link Fault} instance based on {@link Throwable}.
+	 * @param entityId entity identifier
+	 * @param throwable 
+	 * @param auditSupplementaryData
+	 * @return {@link Fault} instance generated for {@link Throwable}
+	 */
+	public static Fault exceptionToFault(CharSequence entityId, Throwable throwable,
+			Map<CharSequence, CharSequence> auditSupplementaryData) {
+		Fault.Builder faultBuilder = Fault.newBuilder();
+		faultBuilder.setInputObjectId(entityId);
+		faultBuilder.setTimestamp(System.currentTimeMillis());
+		faultBuilder.setCode(throwable.getClass().getName());
+		faultBuilder.setMessage(throwable.getMessage());
+		StringWriter strWriter = new StringWriter();
+		PrintWriter pw = new PrintWriter(strWriter);
+		throwable.printStackTrace(pw);
+		pw.close();
+		faultBuilder.setStackTrace(strWriter.toString());
+		if (throwable.getCause()!=null) {
+			faultBuilder.setCauses(appendThrowableToCauses(
+					throwable.getCause(), new ArrayList<Cause>()));
+		}
+		if (auditSupplementaryData!=null && !auditSupplementaryData.isEmpty()) {
```

Comment:
I don't want to set empty map if it does not contain any values. I am OK with having `null` here rather than having empty map instance serialized by avro.